yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,186 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    ExtractorError,
+    try_get,
+    urlencode_postdata, int_or_none, str_or_none,
+)
+
+
+class FreeTvBaseIE(InfoExtractor):
+    def _get_api_response(self, content_id, postdata, resource_type):
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            content_id, data=urlencode_postdata(postdata), fatal=False,
+            note='Downloading {} {} JSON'.format(content_id, resource_type))
+
+        if response is False:
+            raise ExtractorError("Couldn't get response", expected=True)
+
+        if response.get('data') is False:
+            raise ExtractorError("Response doesn't contain {} data".format(resource_type), expected=True)
+
+        return response
+
+    def _extract_video(self, content_id, action="olyott_video_play"):
+        api_response = self._get_api_response(
+            content_id, resource_type='video',
+            postdata={
+                'action': action,
+                'contentID': content_id,
+            })
+
+        video_id = try_get(api_response, lambda x: x['data']['displayMeta']['contentID'], expected_type=str)
+        title = try_get(api_response, lambda x: x['data']['displayMeta']['title'])
+        description = try_get(api_response, lambda x: x['data']['displayMeta']['desc'])
+        video_url = try_get(api_response, lambda x: x['data']['displayMeta']['streamURLVideo'], expected_type=str)
+
+        ext = determine_ext(video_url)
+        if ext != 'm3u8':
+            raise ExtractorError('No manifest was found', video_id=video_id, expected=True)
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, video_id, 'mp4')
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'description': description,
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+    def _extract_series(self, display_id, season_ids, series_title, series_description,
+                        action="olyott_get_dynamic_series_content"):
+        entries = []
+        for season_id in season_ids:
+            api_response = self._get_api_response(
+                season_id, resource_type='series',
+                postdata={
+                    'action': action,
+                    'contentID': season_id,
+                    'type': 'list',
+                    'perPage': '1000'
+                })
+
+            season_episodes = try_get(api_response, lambda x: x['data']['1'])
+
+            if season_episodes is None:
+                raise ExtractorError("Response doesn't contain episode list", expected=True)
```

Comment:
why expected=True?

Reply:
I thought it would make it fatal 🤔 

Should be fixed now, I've used the default value